src/backend/dx12/src/command.rs

```diff
@@ -1205,6 +1208,15 @@ impl com::CommandBuffer<Backend> for CommandBuffer {
         assert_eq!(self.raw, native::GraphicsCommandList::null());
         self.allocator = Some(allocator);
         self.raw = list;
+
+        if self.raw_name.is_empty() {
+            // Reset the name so a previously named raw GraphicsCommandList's 
+            // name won't be used for an unnamed CommandBuffer.
+            const EMPTY_NAME: u16 = 0;
+            self.raw.SetName(&EMPTY_NAME);
```

Comment:
instead of doing this in `begin()`, did you consider resetting the name unconditionally when the `raw` gets returned to the `pool_shared`? It seems like that could be an elegant solution.

Reply:
I was afraid that graphics debuggers may not pick up the name if it gets cleared "early", but now that I fiddled around a bit with RenderDoc and PIX, concluded that they don't care about `ID3D12GraphicsCommandList` object names at all. Even tried with modified C++ DX12 samples, on multiple vendors, still no name is shown for command list objects. Other object (e.g. `CommandQueue`, `CommandAllocator`) names are shown properly.

So setting the name seems to be useless, as it has no effect. Of course it doesn't hurt doing it anyway.

Reply:
Would you want to file an issue to `RenderDoc` about this?

Reply:
Sure! Probably not an issue, since SetName is explicitly skipped in the injected wrapper code for CommandLists, but maybe the author(s) can shed some light on the reasons.